openfaas/faas-netes
# Move YAML into separate folder with openfaas/fn namespaces

## Description


Move YAML into separate folder with openfaas/fn namespaces

## Motivation and Context

This brings us closer into parity with how we suggest deploying via `helm`

For OpenShift a specific namespace is required other than default and helm is difficult for them to use with the security setup so this should help them too.

Asynchronous processing is in by default and can be turned off by manually editing/removing NATS Streaming. @stefanprodan suggests you'd always want this enabled.

Also merges: #52 and #85 

## How Has This Been Tested?




On a 2-node `kubeadm` cluster with `rbac` enabled

Clear the cluster down:

```
$ kubectl delete deploy --all
$ kubectl delete svc --all
```

Test like this:

```
$ kubectl apply -f ./namespaces.yml,./yaml
```

Then go through all the normal flows such as:

* Create function
* Use CLI
* Deploy via store
* Invoke with load to trigger scaling
* Check Prometheus stats
* Invoke asynchronously - check you get a result.
* How is the latency before/after applying the new config? Better/worse/indifferent?

## Types of changes


## Checklist:



Requires change to blog posts / guides / deployments.
